### DISCUSSION TITLE
Shelly RGBW2 and CCT LED Stripe with only 2 connectors

### DISCUSSION THREAD

**Discussion by Freasenx** [arendst/Tasmota#17849] [Q&A]
Hello,
I have two Shelly RGBW2. Both I have flashed with Tasmota.
One of them I have connected to a CCT LED Stripe. This LED stripe has a plus pole which I have bridged with the plus pole of the Shelly and one each minus pole for warm white and cold white. These go into the G and R port. So together three connections

After setting the PWMs in the Tamplate, dimming and changing the temperature works without problems.

The second LED however has a plus pole and a minus pole which according to the LED is responsible for warm white and cold white. 
The LEDs are with the original controller also dimmable
together two connections

How exactly do I connect this to a Shelly RGBW2 and what exactly do I have to set in Tasmota for this?

**Comment by sfromis** [comment]
What exact LED strip 2 do you have, and where's the info on it?

**Comment by Freasenx** [comment]
From my Nobila kitchen. Derby II Emotion 
![IMG_0609](https://user-images.githubusercontent.com/119683072/216079368-b9ff880c-7f2a-42ef-ae59-277eae835ee5.jpeg)

**Reply by sfromis** [reply]
You did not post any info in the strip, or even show how the connections are. From the image, lt looks like it might be 4 wires, not just two. You may have two cables with two wires in each?

Or they may be using variable voltages, instead of "normal" PWM control....

**Reply by Freasenx** [reply]
This is the original controller. Each LED strip is connected to it.

Each LED Strip has 2 connections and 1 cable. What information is missing? I can try to find out

![image](https://user-images.githubusercontent.com/119683072/216083022-a3699edc-8f5c-4c7b-8af1-2450bde86943.png)
![IMG_0611](https://user-images.githubusercontent.com/119683072/216083853-ed331249-23c2-4d58-a17b-ef8144c0f7f5.jpeg)

**Reply by Freasenx** [reply]
Is it possible that the LEDs become cold or warm depending on which connection is positive or negative?

For example
Connection 1 Plus
Connection 2 minus
Cold white

Connection 1 Minus
Connection 2 Plus
Warm white

**Reply by barbudor** [reply]
This type of led strip requires a H-bridge
This is supported by Tasmota but not by the Shelly
You need a specific harware for that

**Reply by barbudor** [reply]
I haven't found a ready commercial device that provides H-Bridge (it may exist, I just haven't found one) but it is something that you can DIY yourself if you are confident with electronics tinkering
You would need:
- an ESP32 based board
- a H-bridge module

A H-bridge is typicaly a DC motor driver which allows to control the direction of the motor by reverting the polarity. Exactly what you need for changing the polarity of your led strip.
This module should work : https://www.aliexpress.com/item/1005004546701788.html
It is a dual H-bridge meant for motors but it should work for leds. You will use only half of it (only 1 channel but it is dual channel)

You would also need a ESP32 module because H-bridge is only supported on ESP32
One like this one would work: https://www.aliexpress.com/item/1005004647945223.html

These are examples, there are many others available on AliExpress or Amazon or other shops
If you want to power your ESP32 board from hte same power supply as the Led, you will need an 12VDC to 5VDC converter like: 
- this one but chose the right model (5V outut) : https://www.aliexpress.com/item/1005001597589461.html
- this one which is adjustable so you will have to tune to have 5V : https://www.aliexpress.com/item/32653212622.html

Let us know if you are ready going this way and need more help

**Reply by Freasenx** [reply]
Thanks for the detailed description, that helps me a lot. H-Bridge was simply not a name for me. I try myself on the whole thing and see if I can do it !

**Comment by Noschvie** [comment]
Can this [ESP32 module](https://www.robot-electronics.co.uk/wled03.html) be used to control such CCT LED stripes ? Thanks!

**Reply by barbudor** [reply]
Unfortunately not. This module is pretty much equivalent to the Shelly

If you look at the schematic of the ESP32 module you link you will see that there is only 1 transistor per led
See that transistor as a fast relay connecting the led strip to ground. Which mean that such module can only control the current flow in one single direction

This is the principle diagram for a h-bridge
![image](https://user-images.githubusercontent.com/2996042/216322873-438ff41d-f237-4f5b-be41-4802996a5529.png)
There you have 4 transistors. Generally that goes with a controller which insure that at a given time you have only
- top left ans bottom right transistors conducting
- Or bottom left and top right

But never should 2 transistor of the side conduct at the same side or magic smoke will happen

As you see, depending which pair is conducting, the motor or your led strip, would see the current going from left to right or right to left

This is what you need and unfortunately I don't see another way than building it yourself if you want to replace the existing controller. 

Now, we can look at possible alternative
How is this controller driven? 
Does it have a serial port? 
Or can you connect external potentiometer? 

It might be simpler to keep using that working controller but use Tasmota to drive it
Could you share details about the controller?

**Reply by barbudor** [reply]
> It might be simpler to keep using that working controller but use Tasmota to drive it

I tried to look for details on the controller but there doesn't seem to be an easy way to hack it
It is meant to be controlled by RF remote of the brand or their "home gateway" which also use RF
There are 2 small connectors on the side for switches or sensors sur as PIR motion sensors
There seems to some kind of slot for modules on the other side but I've found no details on that

**Reply by Freasenx** [reply]
I have done a little research on the system. It does not look like you can flash Tasmota on it.

But there is the Emotion Home Base. This is Zigbee compatible. So I build the connection to my SmartHome. Thanks for your support !

**Comment by mikeoh90** [comment]
Anyone coming across this same issue now, Gledopto have a zigbee module that seems to support 2 wire CCT strips
